src/main/java/keycloak/scim_user_spi/SCIMUserStorageProvider.java

```diff
@@ -158,9 +157,17 @@ public boolean isValid(RealmModel realm, UserModel user, CredentialInput input)
 		if (!supportsCredentialType(input.getType()) || !(input instanceof UserCredentialModel)) return false;
 
 		UserCredentialModel cred = (UserCredentialModel)input;
-		String password = properties.getProperty(user.getUsername());
-		if (password == null) return false;
-		return password.equals(cred.getValue());
+		/* The password can either be validated locally in keycloak (tried first)
+		   or in the SCIM server */
+		if (session.userCredentialManager().isConfiguredLocally(realm, user, input.getType())) {
+			logger.debug("Local password validation for " + user.getUsername());
+			// return false in order to fallback to the next validator
+			return false;
+		} else {
+			logger.debug("Delegated password validation for " + user.getUsername());
+			Scim scim = this.scim;
+			return scim.isValid(user.getUsername(), input.getChallengeResponse());
```

Comment:
Returning false here will trigger the fallback to the local keycloak password validation, so I don't believe the first branch of this `if` conditional is needed?